### PR TITLE
feat(dot/parachain): Implement candidate backing skeleton

### PR DESCRIPTION
**Pull request by axaysagathiya**
## Tests

```sh
go test -tags integration github.com/ChainSafe/gossamer
```

## Primary Reviewer

@timwu20 @kishansagathiya

### CODE REVIEW

**Review by kishansagathiya — COMMENTED** [review]
LGTM, but lets create individual issues for all the TODOs, before we merge this.

**Review by kishansagathiya — APPROVED** [review]

File: dot/parachain/backing/candidate_backing.go [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+	SignedFullStatement SignedFullStatementWithPVD
+}
+
+func New(overseerChan chan<- any) *CandidateBacking {
+	return &CandidateBacking{
+		SubSystemToOverseer: overseerChan,
+	}
+}
+
+func (cb *CandidateBacking) Run(ctx context.Context, OverseerToSubSystem chan any, SubSystemToOverseer chan any) error {
```
Comment (EclesioMeloJunior) [inline dot/parachain/backing/candidate_backing.go:65]:
```suggestion
func (cb *CandidateBacking) Run(ctx context.Context, overseerToSubSystem chan any, subSystemToOverseer chan any) error {
```

File: dot/parachain/available_data_fetching.go [hunk 2]
```diff
@@ -57,6 +57,7 @@ type AvailableData struct {
 	ValidationData PersistedValidationData `scale:"2"`
 }
 
+// TODO: remove this type, copied to dot/parachain/types
```
Comment (EclesioMeloJunior) [inline dot/parachain/available_data_fetching.go:60]: is it a big refactor removing this type in this PR?

**Review by EclesioMeloJunior — APPROVED** [review]
LGTM, just a comment over parameter names

File: dot/parachain/available_data_fetching.go (hunk 2 above)
Comment (axaysagathiya) [reply dot/parachain/available_data_fetching.go]: Nope, I removed it.

**Pull request merged**